frontend/src/metabase/query_builder/components/view/View.jsx

```diff
@@ -350,7 +350,10 @@ class View extends Component {
       );
     }
 
-    if (question.isDataset() && queryBuilderMode === "dataset") {
+    const isModelOrMetric =
+      question.type() === "model" || question.type() === "metric";
```

Comment:
Did you intentionally avoid using `question.type() !== "question"` for the sake of being more explicit?

Reply:
Also because `question.type()` can be `undefined`

Reply:
I thought that we're falling back to "question" if it's undefined.
But yeah, until we have the type safety here, this makes sense.

Thank you